### PR TITLE
build codesign cipd pkg

### PR DESCRIPTION
**Pull request by XilaiZhang**
build cipd package of codesign standalone.

context:
standalone prs: https://github.com/flutter/cocoon/pull/1861, 1899, 1900
design doc: go/code-signing-bot, go/code-signing-standalone

### CODE REVIEW

**Comment by XilaiZhang** [comment]
cc @CaseyHillers but feel free to ignore since go/chillers-ooo

File: codesign/tool/ensure_file [hunk 1]
```diff
@@ -0,0 +1,3 @@
+$ServiceURL https://chrome-infra-packages.appspot.com/
+
+dart/dart-sdk/${os}-${arch} beta
```
Comment (keyonghan) [inline codesign/tool/ensure_file:3]: ditto.

File: codesign/tool/build.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+# Copyright 2019 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+# Fetches corresponding dart sdk from CIPD for different platforms, builds
+# an executable binary of device_doctor to `build` folder.
+#
+# This currently supports linux and mac.
```
Comment (keyonghan) [inline codesign/tool/build.sh:9]: Do we need to support linux for codesign tool?

**Review by keyonghan — COMMENTED** [review]
Can we also update the README to include how to build and how to use?

File: codesign/tool/build.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  exit -1
+fi
+
+mkdir -p build
+tool/dart-sdk/bin/dart pub get
+tool/dart-sdk/bin/dart compile exe bin/main.dart -o build/codesign
+
+cp -f LICENSE build/
+
+popd
```
[the remark this answers is not in the document]
Comment (XilaiZhang) [reply codesign/tool/build.sh:36]: Umm I thought we don’t need an ending newline? Or maybe we need it for some reasons

File: codesign/tool/build.sh [hunk 4]
```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+# Copyright 2019 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+# Fetches corresponding dart sdk from CIPD for different platforms, builds
+# an executable binary of device_doctor to `build` folder.
```
[the remark this answers is not in the document]
Comment (XilaiZhang) [reply codesign/tool/build.sh:7]: Sure, will do

File: codesign/tool/build.sh (hunk 2 above)
Comment (XilaiZhang) [reply codesign/tool/build.sh:9]: Let me update this

File: codesign/README.md [hunk 5]
```diff
@@ -1 +1,53 @@
...
+When a new commit is submitted, post-submit builders will trigger a new version
+with a tag of `commit_sha`, and a ref of `staging`.
+
+### Manual build
+
+Running `tool/build.sh` will build an executable binary in
+the `build` folder. Then push to cipd by running
+
+```bash
+cipd create -in build                   \
```
Comment (XilaiZhang) [inline codesign/README.md:30]: I uploaded a binary but it doesn't show up on https://chrome-infra-packages.appspot.com/p/flutter/codesign yet. Maybe it takes some extra time to propagate

**Pull request merged**